### PR TITLE
Implements program section for unscheduled events

### PR DESCRIPTION
**Pull request by ChrisBr**
As long as the schedule is not public an overview of the confirmed events is shown instead.
Links to the program are on the index page and the conference splash page.

Program

### CODE REVIEW

**Comment by ChrisBr** [comment]
@hennevogel @differentreality please review

@hcderaad

File: app/models/event.rb [hunk 1]
```diff
@@ -209,7 +209,7 @@ def speaker_names
     speakers.each do |speaker|
       result.push(speaker.name)
     end
-    result.to_sentence
+    result.uniq.to_sentence
```
Comment (hennevogel) [inline app/models/event.rb]: Please use a [Ruby set](http://ruby-doc.org//stdlib-2.1.1//libdoc/set/rdoc/Set.html) here.

File: app/views/conference/_program_splashpage.html.haml [hunk 2]
```diff
@@ -23,7 +23,12 @@
         %p.cta-button.text-center
           = link_to(schedule_conference_path(@conference.short_title), class: 'btn btn-default btn-lg') do
             Full Schedule
-
+  - elsif @conference.events.confirmed.any?
+    .row
+      .col-md-12
+        %p.cta-button.text-center
+          = link_to(schedule_conference_path(@conference.short_title), class: 'btn btn-default btn-lg') do
+            View confirmed events
```
Comment (hennevogel) [inline app/views/conference/_program_splashpage.html.haml]: No visitor knows what that means.

**Comment by hennevogel** [comment]
How do I not show the confirmed events?

File: app/models/event.rb (hunk 1 above)
Comment (ChrisBr) [inline app/models/event.rb]: Okay

File: app/views/conference/_program_splashpage.html.haml (hunk 2 above)
Comment (ChrisBr) [inline app/views/conference/_program_splashpage.html.haml]: Do you have a better name? Maybe just "Program"?

**Comment by ChrisBr** [comment]
@hennevogel That's not possible! 
- If there are confirmed events the program page is displayed
- If the schedule is ready and public the schedule is displayed
- Otherwise the schedule page is not visible at all

Any concerns with this workflow?

**Comment by hennevogel** [comment]
Yeah sure. What if I don't want to reveal my program before I have a schedule ready?

**Comment by ChrisBr** [comment]
@hennevogel Okay, what do you suggest? A boolean flag in cfp like we did it for schedule public?

**Comment by hennevogel** [comment]
I'm still a fan of integrating schedule and listing and control it via the setting that is already there.

**Comment by ChrisBr** [comment]
But then you have an empty schedule displayed which is
- ugly 
- and maybe misleading

File: app/views/conference/_program_splashpage.html.haml (hunk 2 above)
Comment (hennevogel) [inline app/views/conference/_program_splashpage.html.haml]: Schedule? ;-)

**Comment by hennevogel** [comment]
What's the problem with catching this case and displaying nothing?

**Comment by ChrisBr** [comment]
@hcderaad sorry for the delay ...

No scheduled events:
Program


No scheduled events:
Program

With scheduled events:
Program

@hennevogel @differentreality please review

**Comment by hcderaad** [comment]
@ChrisBr  No problem, thanks a lot for your efforts, this looks very usable! Looking forward to see this added when done!

**Comment by differentreality** [comment]
When we have a schedule, we should only show extra content (confirmed, not scheduled) in the other tab, right? So perhaps Program tab should be named *More content* ?

On a sidenote, we should stick to the usage of a single word in general. Splashpage configuration says **include program**', when it should really say **'include schedule**.

**Comment by hennevogel** [comment]
Don't make it too complicated. It's not "extra content" it's another view of our events.

**Comment by ChrisBr** [comment]
@hennevogel @differentreality So we will stay with Program?

@differentreality For the second point I agree with you!

**Comment by hennevogel** [comment]
LGTM :+1:

**Comment by hcderaad** [comment]
@ChrisBr @hennevogel @differentreality  Thank you all for your effort in implementing this feature!

**Pull request merged**